src/ObjectBuilder.cs

```diff
@@ -92,12 +96,24 @@ public virtual IModelType BuildServiceType(string serviceTypeName, bool required
                             var value = valueOverride["value"];
                             var description = valueOverride["description"];
                             var name = valueOverride["name"] ?? value;
-                            enumType.Values.Add(new EnumValue
+                            
+                            var enumVal = new EnumValue
                             {
                                 Name = (string)name,
                                 SerializedName = (string)value,
                                 Description = (string)description
-                            });
+                            };
+
+                            // set the allowedValues if any
+                            if(valueOverride["allowedValues"]!=null)

```

Comment:
PR comment not really addressed, the point is that you can combine this check with the stuff in foreach 😉 see other comment